src/boot/parser.mly

```diff
@@ -270,10 +272,15 @@ left:
 
 
 atom:
-  | atom DOT label       { TmProj(mkinfo (tm_info $1) $2.i, $1, $3) }
-  | LPAREN seq RPAREN    { if List.length $2 = 1 then List.hd $2
-                           else TmTuple(mkinfo $1.i $3.i,$2) }
-  | LPAREN RPAREN        { TmConst($1.i, Cunit) }
+  | atom DOT proj_label
+      { let fi = mkinfo (tm_info $1) (fst $3) in
+        let id = unique_ident in
+        TmMatch(fi,$1,PatRecord(fi,Record.singleton (snd $3) (PatNamed(fi,NameStr(id,nosym)))),
+                                TmVar(fi,id,nosym), TmNever(fi)) }
+  | LPAREN seq RPAREN
+      { if List.length $2 = 1 then List.hd $2
+        else tuple2record (mkinfo $1.i $3.i) $2 }
+  | LPAREN RPAREN        { TmRecord($1.i, Record.empty) }
```

Comment:
This means we can write unit both as `()` and `unit` (and `{}`)

Reply:
Yes, we can write `()` or `{}` because they are the same thing. We have for historical reasons also been able to write `unit`. I would vote for remove `unit`, but let's discuss this at the Miking meeting. Several files use `unit` today.